prov/verbs/src/ep_rdm/verbs_av_ep_rdm.c

```diff
@@ -607,6 +616,7 @@ fi_ibv_rdm_av_tbl_idx_to_conn_add_new_conn(struct fi_ibv_rdm_ep *ep,
 			HASH_ADD(hh, av_entry->conn_hash, ep,
 				 sizeof(struct fi_ibv_rdm_ep *), conn);
 		}
+		pthread_mutex_unlock(&av_entry->conn_lock);
 	}
 
```

Comment:
Can we restructure the error handling to jump (i.e. goto) a common exit routine, so that we don't miss calling unlock anywhere, including possible future updates to the code?

Reply:
Yes, we can do this.
I've updated the places where it's possible